lib/Sema/CSDiagnostics.cpp

```diff
@@ -2041,6 +2042,37 @@ bool AllowTypeOrInstanceMemberFailure::diagnoseAsError() {
         emitDiagnostic(loc, diag::super_initializer_not_in_initializer);
         return true;
       }
+
+      // Dig through the chain of member accesses until we find the last
+      // UnresolvedDotExpr.
+      auto *lastUnresolved = ctorRef;
+      while (auto *nextMember =
+                 dyn_cast_or_null<UnresolvedDotExpr>(lastUnresolved->getBase()))
+        lastUnresolved = nextMember;
+
+      if (auto *DRE = dyn_cast<DeclRefExpr>(lastUnresolved->getBase())) {
```

Comment:
I think instead of using expressions here you will be better of retrieving choices made for base from constraint system, this way even if the base was overloaded it would be easy to determine whether it's settable.

Reply:
Makes sense. I'll update this with that change in a bit.